### DISCUSSION TITLE
Is it possible to use workspace libraries in build-time tooling?

### DISCUSSION THREAD

**Discussion by bradbarrow** [nrwl/nx#5589] [General]
Hi folks, I'm new to using nx and was wondering whether it's possible to consume libraries generated with `nx` in the build time tooling of apps in the workspace.

For e.g. I've created a NextJS app, and I want to do some work in the `next.config.js` file. Let's imagine I've got a helper function that does that:

```
// next.config.js

const withAlteredNextJSConfig = (config) => {
  config.altered = true;

  return config;
};

module.exports = withAlteredNextJSConfig({});
```

I'm going to want to do the same things for any other NextJS app I build. So my thought is to make a package:

```
const withAlteredNextJSConfig = require('@my-workspace/alter-next-config'); // an nx generated library

module.exports = withAlteredNextJSConfig({});
```

Running this results in `Module not found @my-workspace/alter-next-config`

Makes sense, as that package doesn't actually exist. 

So my question is:

> Is there any way to consume nx generated libs in the build time tooling of apps in the workspace, as per the example above?

Thanks all :)

**Comment by rrooding** [comment]
I would also like to know if this is possible in any way

**Comment by brandonroberts** [comment]
Yea, you could do this. In order to use the library path from within a `.js` file, you'll need to use [tsconfig-paths](https://www.npmjs.com/package/tsconfig-paths) to access the path aliases in the `tsconfig.base.json`

**Reply by thiniulian** [reply]
Hello! Do you have any examples or pointers on how to do this?
